Kernel/TTY/TTY.h

```diff
@@ -40,14 +41,28 @@ class TTY : public CharacterDevice {
 
     TTY(unsigned major, unsigned minor);
     void emit(u8);
+    virtual void echo(u8);
```

Comment:
Can we make this a pure virtual instead:
```suggestion
    virtual void echo(u8) = 0;
```
That would make it an error to subclass `TTY` without overriding `echo(u8)`, which seems like a reasonable requirement.